robusta_krr/core/integrations/kubernetes.py

```diff
@@ -26,6 +28,8 @@ def __init__(self, cluster: Optional[str], *args, **kwargs):
         super().__init__(*args, **kwargs)
 
         self.cluster = cluster
+        # This executor will be running requests to Kubernetes API
+        self.executor = ThreadPoolExecutor(6)
```

Comment:
This could also make the scan take longer, krr_scan times out after 5 minutes, we might want to run some tests before we merge to get an idea for the time increase that we will have now or we could break the action

Reply:
So, by default `await asyncio.to_thread()` runs the syncronous task in a default threadpool, being limited to `min(32, os.cpu_count() + 4)`.

So, a node with lots of cores will have max of 32 threads opened.

I changed that behaviour to limit the size of a threadpool, so limiting the amount of requests is also provided that way.